vault/resource_identity_group.go

```diff
@@ -110,26 +110,41 @@ func identityGroupResource() *schema.Resource {
 	}
 }
 
-func identityGroupUpdateFields(d *schema.ResourceData, data map[string]interface{}) error {
-	if name, ok := d.GetOk("name"); ok {
-		data["name"] = name
-	}
+func identityGroupUpdateFields(d *schema.ResourceData, data map[string]interface{}, create bool) error {
+	if create {
+		if name, ok := d.GetOk("name"); ok {
+			data["name"] = name
+		}
 
-	if externalPolicies, ok := d.GetOk("external_policies"); !(ok && externalPolicies.(bool)) {
-		data["policies"] = d.Get("policies").(*schema.Set).List()
-	}
+		if externalPolicies, ok := d.GetOk("external_policies"); !(ok && externalPolicies.(bool)) {
+			data["policies"] = d.Get("policies").(*schema.Set).List()
+		}
 
-	// Member groups and entities can't be set for external groups
-	if d.Get("type").(string) == "internal" {
-		data["member_group_ids"] = d.Get("member_group_ids").(*schema.Set).List()
+		// Member groups and entities can't be set for external groups
+		if d.Get("type").(string) == "internal" {
+			data["member_group_ids"] = d.Get("member_group_ids").(*schema.Set).List()
 
-		if externalMemberEntityIds, ok := d.GetOk("external_member_entity_ids"); !(ok && externalMemberEntityIds.(bool)) {
-			data["member_entity_ids"] = d.Get("member_entity_ids").(*schema.Set).List()
+			if externalMemberEntityIds, ok := d.GetOk("external_member_entity_ids"); !(ok && externalMemberEntityIds.(bool)) {
+				data["member_entity_ids"] = d.Get("member_entity_ids").(*schema.Set).List()
+			}
 		}
-	}
 
-	if metadata, ok := d.GetOk("metadata"); ok {
-		data["metadata"] = metadata
+		if metadata, ok := d.GetOk("metadata"); ok {
+			data["metadata"] = metadata
+		}
+	} else {
+		if d.HasChanges("name", "external_policies", "policies", "metadata") {
```

Comment:
Looks like `member_entity_ids` and `member_group_ids` are missing from this function call. They will also need to be set below.